// @onedoc/client/dist/index.mjs
// src/htmlBuilder.ts
var HtmlBuilder = class {
  title;
  start = `<!DOCTYPE html>
                             <html  lang="en">
                                <head>
                                    <meta charset = "UTF-8">
                                    <meta name="viewport" content="width=device-width">`;
  middle = `
                                </head>
                                <body>`;
  end = `</body>
                        </html>`;
  constructor(title) {
    this.title = title;
  }
  build(document, styleSheets) {
    if (styleSheets) {
      styleSheets.forEach((path) => {
        this.start += `<link rel = "stylesheet" href=${path} />`;
      });
    }
    if (this.title) {
      this.start += `<title>${this.title}</title>`;
    }
    this.middle += document;
    return this.start + this.middle + this.end;
  }
};

// src/client.ts
import mime from "mime-types";
var Onedoc = class {
  apiKey;
  endpoint;
  constructor(apiKey, endpoint = "https://api.fileforge.com") {
    this.apiKey = apiKey;
    this.endpoint = endpoint;
  }
  buildUrl(path) {
    return `${this.endpoint}${path}`;
  }
  async render(document) {
    const assets = document.assets ?? [];
    assets.push({ path: "/index.html", content: document.html });
    const test = document.test ?? true;
    const save = document.save ?? false;
    const expires_in = document.expiresIn ?? 1;
    const expires_at = new Date(
      Date.now() + expires_in * 24 * 60 * 60 * 1e3
    );
    const expires_at_iso = expires_at.toISOString();
    const formData = new FormData();
    const optionsBlob = new Blob(
      [
        JSON.stringify({
          test,
          host: save,
          expiresAt: expires_at_iso,
          fileName: document.title ?? "document"
        })
      ],
      { type: "application/json" }
    );
    formData.append("options", optionsBlob);
    assets.forEach((asset) => {
      if (asset.path === "/index.html" && asset.content) {
        const htmlBuilder = new HtmlBuilder(document.title);
        const styleSheets = assets.filter((a) => a.path.endsWith(".css")).map((a) => a.path);
        const html = htmlBuilder.build(document.html, styleSheets);
        const fileBlob = new Blob([html], { type: "text/html" });
        formData.append("files", fileBlob, "index.html");
      } else if (asset.content) {
        const assetType = mime.lookup(asset.path) || "application/octet-stream";
        const fileBlob = new Blob([asset.content], { type: assetType });
        formData.append("files", fileBlob, asset.path);
      }
    });
    const response = await fetch(
      await this.buildUrl("/pdf/generate"),
      {
        method: "POST",
        headers: {
          "x-api-key": this.apiKey
        },
        body: formData
      }
    );
    if (response.status === 201) {
      if (!save) {
        return {
          file: await response.arrayBuffer(),
          link: null,
          error: null,
          info: {}
        };
      } else {
        const jsonResponse = await response.json();
        return { file: null, link: jsonResponse.url, error: null, info: {} };
      }
    } else {
      const error = await response.json();
      return { file: null, link: null, error, info: {} };
    }
  }
};
export {
  HtmlBuilder,
  Onedoc
};
